src/stores/appStore.ts

```diff
@@ -9,101 +8,63 @@ import {
-            const allGeojson = await getGlobalGeojsonFeatures(1);
+            const state = this as AppState;
+            state.appConfig = await getAppConfig();
+            state.countryBoundingBoxes = await getCountryBoundingBoxes();
+            state.admin1Geojson = await getGlobalGeojsonFeatures(1);
+            state.admin1Indicators = await getGlobalIndicators(1);
 
-            this.countryBoundingBoxes = await getCountryBoundingBoxes();
-            this.admin1Geojson = allGeojson;
-            this.admin1Indicators = allIndicators;
+            state.initialisationComplete = true;
         },
-        async selectCountry(countryId: string) {
-            if (countryId === this.selectedCountryId) {
-                return;
-            }
+        async updateMapSettings(newMapSettings: MapSettings) {
+            const state = this as AppState;
+            const { country, adminLevel } = newMapSettings;
 
-            if (!countryId) {
-                this.selectedCountryId = "";
-                this.admin0GeojsonFeature = null;
+            if (!country) {
+                state.mapSettings = newMapSettings;
+                state.admin0GeojsonFeature = null;
                 return;
             }
 
+            let level = adminLevel;
             // Some countries do not have admin2 regions or data - if one of these is selected, we load
             // a more detailed geojson, and re-use its admin1 indicators as "admin2"
-            const level = this.appConfig.countriesWithoutAdmin2.includes(countryId) ? 1 : 2;
+            const admin2DataMissing = state.appConfig.countriesWithoutAdmin2.includes(country);
+            if (level === 2 && admin2DataMissing) {
+                level = 1;
+                state.admin2Indicators[country] = state.admin1Indicators[country];
+            }
 
-            if (!(countryId in this.admin2Indicators)) {
-                if (level === 1) {
-                    this.admin2Indicators[countryId] = this.admin1Indicators[countryId];
-                } else {
-                    this.admin2Indicators[countryId] = await getIndicators(countryId, level);
-                }
+            if (level === 2 && !admin2DataMissing && !(country in state.admin2Indicators)) {
+                state.admin2Indicators[country] = await getIndicators(country, level);
             }
-            if (!(countryId in this.admin2Geojson)) {
-                this.admin2Geojson[countryId] = await getGeojsonFeatures(countryId, level);
+
+            if (!(country in state.admin2Geojson)) {
+                state.admin2Geojson[country] = await getGeojsonFeatures(country, level);
             }
 
-            this.admin0GeojsonFeature = (await getGeojsonFeatures(countryId, 0))[0];
-            this.selectedCountryId = countryId;
+            state.admin0GeojsonFeature = (await getGeojsonFeatures(country, 0))[0];
+            // need to account for the fact that some countries do not have admin level 2 data
```

Comment:
Is this a TODO comment, or what does it mean?

Reply:
that comment was supposed to explain the next line, i change the admin level in the `finalMapSettings` because i am accounting for the countries that do not have admin level 2 data